packages/ui/src/views/repo/components/branch-selector/branch-selector.tsx

```diff
@@ -1,61 +1,58 @@
 import { FC } from 'react'
 
 import { Button, DropdownMenu, DropdownMenuTrigger, Icon, Text } from '@/components'
-import { TranslationStore } from '@/views'
-import { cn } from '@utils/cn'
+import { IBranchSelectorStore, TranslationStore } from '@/views'
 
-import { BranchSelectorDropdown, type BranchSelectorDropdownProps } from './branch-selector-dropdown'
+import { BranchSelectorDropdown } from './branch-selector-dropdown'
 
-interface BranchSelectorProps extends BranchSelectorDropdownProps {
-  size?: 'default' | 'sm'
-  prefix?: string
-  className?: string
+interface BranchSelectorProps {
+  useBranchSelectorStore: () => IBranchSelectorStore
   useTranslationStore: () => TranslationStore
 }
+export const BranchSelector: FC<BranchSelectorProps> = ({ useBranchSelectorStore, useTranslationStore }) => {
+  const {
+    selectedBranch,
+    branchList,
+    tagList,
+    size = 'default',
+    setSelectedBranch: onSelectBranch,
+    prefix,
+    repoId,
+    spaceId
+  } = useBranchSelectorStore()
 
-export const BranchSelector: FC<BranchSelectorProps> = ({
-  selectedBranch,
-  branchList,
-  tagList,
-  size = 'default',
-  onSelectBranch,
-  prefix,
-  className,
-  repoId,
-  spaceId,
-  useTranslationStore
-}) => {
-  const isTag = tagList.some(tag => tag.name === selectedBranch.name && tag.sha === selectedBranch.sha)
+  const isTag = selectedBranch
+    ? tagList.some(tag => tag.name === selectedBranch.name && tag.sha === selectedBranch.sha)
+    : false
 
   return (
     <DropdownMenu>
       <DropdownMenuTrigger asChild>
         <Button
-          className={cn(
-            'data-[state=open]:border-borders-8 [&_svg]:data-[state=open]:text-foreground-1 flex items-center gap-1.5 overflow-hidden px-3',
-            className
-          )}
+          className="data-[state=open]:border-borders-8 [&_svg]:data-[state=open]:text-foreground-1 flex items-center gap-1.5 overflow-hidden px-3"
           variant="outline"
           size={size}
         >
           {!prefix && (
             <Icon className="shrink-0 fill-transparent text-icons-9" name={isTag ? 'tag' : 'branch'} size={12} />
           )}
           <Text className="w-full text-foreground-8" truncate align="left">
-            {prefix ? `${prefix}: ${selectedBranch.name}` : selectedBranch.name}
+            {prefix ? `${prefix}: ${selectedBranch?.name || ''}` : selectedBranch?.name || ''}
           </Text>
           <Icon className="chevron-down text-icons-2" name="chevron-down" size={10} />
         </Button>
       </DropdownMenuTrigger>
-      <BranchSelectorDropdown
-        branchList={branchList}
-        tagList={tagList}
-        selectedBranch={selectedBranch}
-        onSelectBranch={onSelectBranch}
-        repoId={repoId}
-        spaceId={spaceId}
-        useTranslationStore={useTranslationStore}
-      />
+      {branchList && (
```

Comment:
```suggestion
      {branchList && (
```

I don’t think this check is necessary, as we’ve already handled it internally. If, for some reason, the `branchList` is empty, we’ll display an appropriate message to indicate that.

![Google Chrome 2024-12-05 16 54 04](https://github.com/user-attachments/assets/7766b0b5-dab1-4512-951a-e7763bb4ad1c)